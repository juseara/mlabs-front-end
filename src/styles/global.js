import { createGlobalStyle } from 'styled-components';
export default createGlobalStyle`

@import url('https://fonts.googleapis.com/css?family=Turret+Road&display=swap');

@font-face {
    font-family: Prime Light;
    src: url('../../assets/fonts/prime/Prime-Light.otf');
}

@font-face {
    font-family: Prime;
    src: url('../../assets/fonts/prime/Prime-Regular.otf');
}

*{
    margin: 0;
    padding: 0;
    outline: 0;
    box-sizing: border-box;
 }

html, body, #root{
    height: 100%;
}

body{
    font: 14px  prime sans-serif;
    background: #f7f7f7;
    -webkit-font-smoothing: antialiased !important;
    
}

ul {
    list-style: none;
}

a, a:link, a:visited, a:hover, a:active {
    text-decoration: none;
}

`;