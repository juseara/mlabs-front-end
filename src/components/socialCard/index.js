import React from 'react';
import PropTypes from 'prop-types';
import { FaQuestion } from 'react-icons/fa'
import {Card, Image, Tile, Button, Header, CardLoged, ImageLoged, IconHelp } from './socialCard.styles';

const SocialCard = (props) => {
    
    return(
        
        <>
            {
                !props.profile? 
                    <Card>
                        <div style={{position:"relative"}}>
                            <IconHelp>
                                    <FaQuestion color="#8b8d8f"/>
                            </IconHelp>
                            <Image img={props.image}/>
                        </div>
                        <Tile>{props.title}</Tile>
                        <Button onClick={props.onPressButton}>Adicionar</Button>
                    </Card>
          
                :
                    <CardLoged onClick={props.onPressButton} color={props.primaryColor}>
                        <Header color={props.primaryColor}>
                            <span>{props.profile.name}</span>
                        </Header>
                        <ImageLoged image={props.imageLogged}/>
                    </CardLoged>
                
            }
               
        </>
    )
}

SocialCard.propTypes={
    image:PropTypes.string,
    imageLogged:PropTypes.string,
    title:PropTypes.string,
    onPressButton:PropTypes.func,
    profile:PropTypes.any,
    primaryColor:PropTypes.string
}

SocialCard.defaultProps={
    image:'',
    imageLogged:'',
    title:'social',
    onPressButton:()=>{console.log("TESTE")},
    profile:undefined,
    primaryColor:'#ffff'
}

export default SocialCard