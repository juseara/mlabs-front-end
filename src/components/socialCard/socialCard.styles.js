import styled from 'styled-components';
import { device } from '../../styles/devices'


export const SlotCard = styled.div`
        width:20%;
        height:260px;
        background: #F0F0F0;
        position: relative;
        display:flex;
        flex-direction:column;
        justify-content:space-around;
        align-items:center;


        @media ${device.laptop} {
                width:30%;
        }

        @media ${device.tablet} {
                width:40%;
        }

        @media ${device.mobileS} {
                width:100%;
        }

        @media ${device.mobileM} {
                width:100%;
        }

        @media ${device.mobileL} {
                width:100%;
        }
`;

export const Card = styled.div`
        width:100%;
        height:100%;
        background: #F0F0F0;
        display:flex;
        flex-direction:column;
        justify-content:space-around;
        align-items:center;
        border: 2px dotted #cbcccd;
        padding: 40px 0;
        position: relative;

        *::after{
                contain:'';
                box-sizing: border-box;
                
                
        }

        
       
`;

export const IconHelp = styled.div`
        position: absolute;
        left: 36px;
        margin-left: 45.5px;
        width: 30px;
        height: 30px;
        border: solid 2px #8b8d8f;
        border-radius:30px;
        cursor: pointer;
        display:flex;
        justify-content:center;
        align-items:center;

        :hover {
                background-color: #8b8d8f;
                
                svg {
                        color:#fff !important;
                }
                
        }

        
        
`;

export const Image = styled.div`
        background: url(${props => `${props.image}`}) no-repeat 0 0;
        background-size: contain;
        width: 60px;
        height: 60px;
        border-radius: 100%;
        
`;

export const Title = styled.h3`
        font-family: 'Turret Road';
        font-weight: normal;
        font-size: 20px;
        color: #535353;
        font-size: 20px;
        text-transform: uppercase;
        text-align: center;
        
`;

export const Button = styled.button`
        width: 99px;
        height: 32px;
        border-radius: 3px;
        background: #E02B4A;
        font-family: 'Arial', sans-serif;
        font-weight: bold;
        font-size: 14px;
        color: #FFF;
        line-height: 14px;
        text-align: center;
        border-width:0;

        :hover {
                opacity:0.5;
        }

        :active {
                opacity:1!important;
        }
`;

export const CardLoged = styled.div`
        width:100%;
        height:100%;
        background: ${props => `${props.color}`};
        display:flex;
        flex-direction:column;
        justify-content:flex-start;
        align-items:center;
        transition: all 0.2s ease-out 0s;
        position: relative;
        border: 0;
        z-index:10;

      

        :hover{
                
                width:114%;
                height:114%;
                margin: -7px;
                box-sizing: content-box;
                box-shadow: 0px 0px 10px 2px rgba(0,0,0,0.2);
                z-index: 40;
                height:296px;
               
        }
`;

export const Header = styled.header`
        background: #0000001f;
        width: 100%;
        padding: 15px 20px;

        span{

                font-family: 'Arial', sans-serif;
                color: #fff;
                display: block;
                max-height: 20px;
                overflow: hidden;
                font-size: 14px;
                text-overflow: ellipsis;
                white-space: nowrap;
                overflow: hidden;
        }
`;

export const ImageLoged = styled.div`
        display:flex;
        align-self:flex-start;
        background: url(${props => `${props.image}`}) no-repeat 0 0;
        background-size: contain;
        height: 90px;
        width: 85px;
        margin: 10px 0 0 10px;
`;