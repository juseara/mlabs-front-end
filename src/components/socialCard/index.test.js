import React from 'react'
import { mount, shallow } from 'enzyme';
import SocialCard from './index';
import facebook from '../../assets/social-img/facebook.png';
import facebookLogged from '../../assets/social-img/facebook_logged.png';
import { Card, IconHelp, Image, Title, Button, CardLoged,Header,ImageLoged } from './socialCard.styles'

const functionMock = jest.fn()

let propsCard ={
    profile:undefined,
    image:facebook,
    title:'Facebook',
    onPressButton:functionMock
}

let propsCardLogged = {
    profile:{
        name:'Loja do arlindo',
    },
    primaryColor:"#3A589B",
    imageLogged:facebookLogged,
    image:facebook,
    title:'Facebook',
    onPressButton:functionMock
}


describe('[Component] SocialCard',()=>{
    
    it('Deve renderizar o card sem página vinculada corretamente',()=>{

        const wrapper = mount(<SocialCard  {...propsCard}/>)
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find(Card)).toHaveLength(1);
        expect(wrapper.find(Card).find(IconHelp)).toHaveLength(1);
        expect(wrapper.find(Card).find(Image)).toHaveLength(1);
        expect(wrapper.find(Card).find(Image).props().image).toBe(facebook)
        expect(wrapper.find(Card).find(Title).children().text()).toBe("Facebook")
        expect(wrapper.find(Card).find(Button).props().onClick).toBe(functionMock)

    })

    it('Deve renderizar o card com página vinculada corretamente',()=>{

       
        const wrapper = mount(<SocialCard  {...propsCardLogged}/>)
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find(CardLoged)).toHaveLength(1);
        expect(wrapper.find(CardLoged).find(Header)).toHaveLength(1);
        expect(wrapper.find(CardLoged).find(ImageLoged)).toHaveLength(1);
        expect(wrapper.find(CardLoged).find(ImageLoged).props().image).toBe(facebookLogged)
        expect(wrapper.find(CardLoged).find(Header).children().text()).toBe("Loja do arlindo")
        expect(wrapper.find(CardLoged).props().onClick).toBe(functionMock)

    })

    it('Deve ser chamado a funcao onPressButton ao clicar no botão do card sem página vinculada',()=>{

        const wrapper = mount(<SocialCard  {...propsCard}/>)

        wrapper.find(Card).find(Button).simulate('click')

        expect(functionMock).toHaveBeenCalled()
    })

    it('Deve ser chamado a funcao onPressButton ao clicar no card com página vinculada',()=>{

        const wrapper = mount(<SocialCard  {...propsCardLogged}/>)

        wrapper.find(CardLoged).simulate('click')

        expect(functionMock).toHaveBeenCalled()
    })

})

