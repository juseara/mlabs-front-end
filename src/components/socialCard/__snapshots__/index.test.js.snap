// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] SocialCard Deve renderizar o card com página vinculada corretamente 1`] = `ReactWrapper {}`;

exports[`[Component] SocialCard Deve renderizar o card sem página vinculada corretamente 1`] = `ReactWrapper {}`;
